src/storage/v2/durability/wal.cpp

```diff
@@ -444,6 +446,42 @@ WalDeltaData ReadSkipWalDeltaData(BaseDecoder *decoder, const uint64_t version)
       }
       break;
     }
+    case WalDeltaData::Type::VECTOR_INDEX_CREATE:
+    case WalDeltaData::Type::VECTOR_INDEX_DROP: {
+      if constexpr (read_data) {
+        auto index_name = decoder->ReadString();
+        if (!index_name) throw RecoveryFailure("Invalid WAL data!");
```

Comment:
This is from before, but let's improve the error message. Make some static string with the content like:
"Invalid WAL data! Your durability WAL files somehow got corrupted. Please contact the Memgraph team {{inject_how}}."